src/main/python/sweep.py

```diff
@@ -0,0 +1,132 @@
+import io
+
+from PyQt5.QtWidgets import (
+   QTableView
+)
+from PyQt5.QtCore import (
+    QAbstractTableModel, QModelIndex, QByteArray,
+)
+from PyQt5 import QtCore
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from delegates import (SvgDelegate, ComboBoxDelegate)
+
+
+class SweepTableModel(QAbstractTableModel):
+    def __init__(self, colnames):
+        super().__init__()
+        self.colnames = colnames
+
+    def get_data(self, data=None):
+
+        if data:
+            self._data = data
+        else:
+            self._data = [
+                [1, "abs0", "Long Square", "passed", "None", None],
+                [3, "abs1", "Long Square", "passed", "None", None],
+                [7, "abs2", "Short Square", "failed", "None", "Vm above threshold, Noise above threshold"],
+                [13, "abs3", "Ramp", "passed", "None", None],
+            ]
+
+        for item in self._data:
+            item.append(tmp_mpl_svg(item[0]))
+            item.append(tmp_mpl_svg(item[0]))
+
+    def rowCount(self, parent=None, *args, **kwargs):
```

Comment:
why do these have args and kwargs?

Reply:
my editor (pycharm) complained that otherwise the signature of the function does not match that of the parent method.

Reply:
huh, what do the variadic args do on the parent method?

Reply:
not sure.